core/flyout_base.js

```diff
@@ -520,6 +529,66 @@ Blockly.Flyout.prototype.show = function(xmlList) {
 
   this.reflowWrapper_ = this.reflow.bind(this);
   this.workspace_.addChangeListener(this.reflowWrapper_);
+
+  this.recordCategoryScrollPositions_();
+};
+
+/**
+ * Store an array of with category names and scrollbar positions.
+ * This is used when scrolling the flyout to cause a category to be selected.
+ * @private
+ */
+Blockly.Flyout.prototype.recordCategoryScrollPositions_ = function() {
+  this.categoryScrollPositions = [];
+  for (var i = 0; i < this.buttons_.length; i++) {
+    if (this.buttons_[i].isCategoryLabel_) {
```

Comment:
You're still directly accessing internals of the buttons (`isCategoryLabel_`, `position_`, `text_`).  Please don't.